client/js/components/StatsLeaderboard.vue

```diff
@@ -0,0 +1,359 @@
+<template>
+  <div>
+    <div id="stats-table-upper-surface" class="d-flex align-end">
+      <v-select
+        v-model="selectedMetric"
+        :items="metricChoices"
+        label="Select Metric"
+        class="filter-select mr-4 fit"
+        data-cy="metric-select"
+      />
+      <v-select
+        v-model="selectedWeeks"
+        :items="weeks"
+        label="Select Weeks"
+        data-cy="week-select"
+        class="flex-grow-1"
+        menu-props="data-week-select-menu"
+        multiple
+        chips
+        deletable-chips
+      />
+    </div>
+    <v-data-table
+      :items="tableRows"
+      :headers="tableColumns"
+      :sort-by.sync="sortBy"
+      :loading="loading"
+      :item-class="tableRowClass"
+    >
+      <template #[`item.rank`]="{item, value}">
+        <span :data-rank="item.username">{{ value }} </span>
+      </template>
+      <template v-for="week in ['total', ...selectedWeeks]" #[`item.week_${week}`]="{item, value}">
+        <v-tooltip :key="`${item.username}_week_${week}_wins`" top v-if="value">
+          <template #activator="{on, attrs}">
+            <v-chip
+              :key="`${item.username}_week_${week}`"
+              :color="colorForScoreByWeek(week, value)"
+              dark
+              :outlined="['primary', '#000'].includes(colorForScoreByWeek(week, value))"
+              v-bind="{
+                ...attrs,
+                ...dataAttribute(item.username, week),
+              }"
+              v-on="on"
+            >
+              {{ tableCell(item, week) }}
+            </v-chip>
+          </template>
+          {{ playersBeaten(item.username, week) }}
+        </v-tooltip>
+      </template>
+    </v-data-table>
+  </div>
+</template>
+<script>
+const Result = {
+  WON: 1,
+  LOST: 2,
+  INCOMPLETE: -1,
+};
+
+export default {
+  name: 'StatsLeaderboard',
+  props: {
+    loading: Boolean,
+    season: {
+      type: Object,
+      default: null,
+    },
+  },
+  data() {
+    return {
+      sortBy: 'rank',
+      metricChoices: ['Points and Wins', 'Points Only', 'Wins Only'],
+      selectedMetric: 'Points and Wins',
+      weeks: [
+        { text: 'Week 1', value: 1 },
+        { text: 'Week 2', value: 2 },
+        { text: 'Week 3', value: 3 },
+        { text: 'Week 4', value: 4 },
+        { text: 'Week 5', value: 5 },
+        { text: 'Week 6', value: 6 },
+        { text: 'Week 7', value: 7 },
+        { text: 'Week 8', value: 8 },
+        { text: 'Week 9', value: 9 },
+        { text: 'Week 10', value: 10 },
+        { text: 'Week 11', value: 11 },
+        { text: 'Week 12', value: 12 },
+        { text: 'Week 13', value: 13 },
+      ],
+      selectedWeeks: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13],
+    };
+  },
+  computed: {
```

Comment:
This entire computed block is a bit of a jumble and I'm very open to finding ways to cleanup or reorganize it